lib/dialects/abstract/connection-manager.js

```diff
@@ -91,29 +91,31 @@ class ConnectionManager {
    * @private
    * @returns {Promise}
    */
-  _onProcessExit() {
+  async _onProcessExit() {
     if (!this.pool) {
-      return Promise.resolve();
+      return;
     }
 
-    return this.pool.drain().then(() => {
-      debug('connection drain due to process exit');
-      return this.pool.destroyAllNow();
-    });
+    await this.pool.drain();
+    debug('connection drain due to process exit');
+
+    return await this.pool.destroyAllNow();
   }
 
   /**
    * Drain the pool and close it permanently
    *
    * @returns {Promise}
    */
-  close() {
+  async close() {
     // Mark close of pool
-    this.getConnection = function getConnection() {
-      return Promise.reject(new Error('ConnectionManager.getConnection was called after the connection manager was closed!'));
+    this.getConnection = async function getConnection() {
+      throw new Error(
+        new Error('ConnectionManager.getConnection was called after the connection manager was closed!')
```

Comment:
Double `new Error` statement, remove one please

Reply:
this was a bug in my automated conversion of `return Promise.reject`, tomorrow I'll double-check my other PRs to make sure no instances of this mistake escaped testing